app/helpers/automated_tests_client_helper.rb

```diff
@@ -216,6 +216,10 @@ def self.check_user_permission(user, grouping)
     unless user.accepted_groupings.include?(grouping)
       raise I18n.t('automated_tests.error.bad_group')
     end
+    # deadline has not passed
+    if assignment.submission_rule.can_collect_now?
+      raise I18n.t('automated_tests.error.after_due_date')

```

Comment:
I'm not sure why you didn't use the existing key (the one that you changed) instead of adding a new one.